src/librustc_codegen_llvm/attributes.rs

```diff
@@ -312,15 +301,6 @@ pub fn from_fn_attrs(
             // In either case, we mark item as explicitly nounwind.
             false
         }
-    } else {
-        // assume this can possibly unwind, avoiding the application of a
-        // `nounwind` attribute below.
-        //
-        // (But: See comments in previous branch. Specifically, it is
-        // unclear whether there is real value in the assumption this
-        // can unwind. The conservatism here may just be papering over
-        // a real problem by making some UB a bit harder to hit.)
-        true
```

Comment:
In https://github.com/rust-lang/rust/pull/65020 nobody knew why this branch should be needed. And all the tests behave as expected without it. So I think we should remove it.